src/utils/sensors.ts

```diff
@@ -0,0 +1,43 @@
+import type { PointerEvent } from 'react';
+import { PointerSensor } from '@dnd-kit/core';
+
+/**
+ * An extended "PointerSensor" that prevent some
+ * interactive html element(button, input, textarea, select, option...) from dragging
+ */
+export class SmartPointerSensor extends PointerSensor {
+  static activators = [
+    {
+      eventName: 'onPointerDown' as const,
+      handler: ({ nativeEvent: event }: PointerEvent) => {
+        if (
+          !event.isPrimary ||
+          event.button !== 0 ||
+          isInteractiveElement(event.target as Element)
+        ) {
+          return false;
+        }
+
+        return true;
+      },
+    },
+  ];
+}
+
+function isInteractiveElement(element: Element | null) {
+  const interactiveElements = [
+    'button',
+    'input',
+    'textarea',
+    'select',
+    'option',
+  ];
+  if (
+    element?.tagName &&
+    interactiveElements.includes(element.tagName.toLowerCase())
+  ) {
+    return true;
+  }
+
+  return false;
+}
```

Comment:
Can we create a component that will act as a handle for drag and drop instead of having the whole question draggable?

Reply:
Do you mean like some sort of anchor/spot from which we would drag the question?

Reply:
Yes